iggy/src/compression.rs

```diff
@@ -0,0 +1,159 @@
+use serde::{
+    de::{self, Deserializer, Visitor},
+    Deserialize, Serialize, Serializer,
+};
+use std::{
+    fmt::{Display, Formatter},
+    str::FromStr,
+};
+
+use crate::error::Error;
+
+// for now only those, in the future will add snappy, lz4, zstd (same as in confluent kafka) in addition to that
+// we should consider brotli as well.
+#[derive(Debug, PartialEq, Clone)]
+pub enum CompressionKind {
+    None,
+    Gzip,
+}
+
+impl FromStr for CompressionKind {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "Gzip" | "gzip" => Ok(CompressionKind::Gzip),
+            "None" | "none" => Ok(CompressionKind::None),
+            _ => Err(format!("Unknown compression type: {}", s)),
+        }
+    }
+}
+
+impl CompressionKind {
+    pub fn as_code(&self) -> u8 {
+        match self {
+            CompressionKind::None => 1,
+            CompressionKind::Gzip => 2,
+        }
+    }
+
+    pub fn from_code(code: u8) -> Result<Self, Error> {
+        match code {
+            1 => Ok(CompressionKind::None),
+            2 => Ok(CompressionKind::Gzip),
+            _ => Err(Error::InvalidCommand),
+        }
+    }
+}
+
+impl Display for CompressionKind {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            CompressionKind::None => write!(f, "none"),
+            CompressionKind::Gzip => write!(f, "gzip"),
+        }
+    }
+}
+
+impl Serialize for CompressionKind {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match self {
+            CompressionKind::None => serializer.serialize_str("none"),
+            CompressionKind::Gzip => serializer.serialize_str("gzip"),
+        }
+    }
+}
+
+impl From<CompressionKind> for String {
+    fn from(value: CompressionKind) -> Self {
+        match value {
+            CompressionKind::None => "none".to_string(),
+            CompressionKind::Gzip => "gzip".to_string(),
+        }
+    }
+}
+struct CompressionKindVisitor;
```

Comment:
Is it still needed along with the `Visitor`, given that the `serde` it not used anymore?